components/common/ConnectButton/_WalletButtonWithMenu.tsx

```diff
@@ -1,95 +1,107 @@
+
+export type WalletButtonWithMenuProps = {
+	mode: ConnectButtonProps["mode"];
+	account: GeneralAccount;
+};
 
 export default function WalletButtonWithMenu({
 	mode,
-}: Pick<ConnectButtonProps, "mode">) {
+	account,
+}: WalletButtonWithMenuProps) {
+	const [refillEmailBalance, checkIsAbleToRefillEmailBalance] = useAccountState(
+		(s) => [s.refillEmailBalance, s.checkIsAbleToRefillEmailBalance]
+	);
 	const [menuOpened, menuHandlers] = useDisclosure(false);
-
-	const [disconnOpened, disconnHandlers] = useDisclosure(false);
-
-	const { disconnect } = useDisconnect();
-
-	const { address } = useAccount();
-	const { data, isLoading: isLoadingBalance } = useBalance({
-		addressOrName: address,
-	});
-
-	const { disconnect: disconnCharacter } = useDisconnectCurrentCharacter();
+	const { balance, isLoading: isLoadingBalance } = useAccountBalance();
+	const disconnectModal = useDisconnectModal();
+	const isAbleToRefillEmailBalance = useIntervalMemo(
+		checkIsAbleToRefillEmailBalance
+	);
 
 	return (
-		<>
-			<Modal
-				opened={disconnOpened}
-				onClose={() => disconnHandlers.close()}
-				title="Disconnect Wallet?"
-				onConfirm={() => {
-					disconnCharacter();
-					disconnect();
-				}}
-				confirmText="Disconnect"
-				confirmType="danger"
-			>
-				<Text>You can always log back in at any time.</Text>
-			</Modal>
-
-			<Menu
-				opened={menuOpened}
-				onOpen={() => menuHandlers.open()}
-				onClose={() => menuHandlers.close()}
-				position="bottom-start"
-				radius="md"
-				transition="scale-y"
-				width="target"
-			>
-				<Menu.Target>
-					<WalletDisplayButton
-						className="w-full"
-						menuOpened={menuOpened}
-						mode={mode}
-					/>
-				</Menu.Target>
+		<Menu
+			opened={menuOpened}
+			onOpen={() => menuHandlers.open()}
+			onClose={() => menuHandlers.close()}
+			position="bottom-start"
+			radius="md"
+			transition="scale-y"
+			width="target"
+		>
+			<Menu.Target>
+				<WalletDisplayButton
+					className="w-full"
+					menuOpened={menuOpened}
+					mode={mode}
+					account={account}
+				/>
+			</Menu.Target>
 
-				<Menu.Dropdown className="w-full">
-					<Menu.Label>$CSB Balance</Menu.Label>
-					<MenuItem>
-						<div className="flex items-center">
-							<Text className="i-csb:logo text-lg" color="black" />
-							<Space w={5} />
-							<Text className="font-600">
-								{isLoadingBalance ? "..." : data?.formatted}
-							</Text>
-						</div>
-					</MenuItem>
+			<Menu.Dropdown className="w-full">
+				<Menu.Label>$CSB Balance</Menu.Label>
+				<Menu.Label>
+					<div className="flex items-center">
+						<Text className="i-csb:logo text-20px text-[#F6C549]" />
+						<Text className="font-400 text-16px ml-5px text-[#082135]">
+							{isLoadingBalance ? "..." : balance}
+						</Text>
+						{account.type === "email" && (
+							<Button
+								size="xs"
+								className="ml-auto h-24px"
+								px={10}
+								radius={6}
+								disabled={!isAbleToRefillEmailBalance}
```

Comment:
Could we show the reason while the mouse hovers on the disabled button? Also there should be "cursor-disabled" style when disabled.